internal/event/clusterevent.go

```diff
@@ -0,0 +1,39 @@
+package event
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+)
+
+// EtcdClusterReconcilerEvent represents the action of the operator having actually done anything. Any meaningful change
+// to the cluster should result in one of these.
+type EtcdClusterReconcilerEvent interface {
```

Comment:
you can probably name this interface something more generic so it can be reused by other events in the operator, e.g. `KubernetesEvent`

Reply:
I don't want to conflate this with a Kubernetes API event. This event can publish an API event, but it's got all sorts of extra stuff on it that the API event does not.

I can see wanting to be more generic. Could just be `ReconcilerEvent`, to emphasise that this event only exists in the reconciler.

Reply:
i don't think having a separate interface for peer events will stop us emitting events for the wrong reconciler. if we create a second interface 
```go
type EtcdPeerReconcilerEvent struct {
	// Record this into an event recorder
	Record(recorder record.EventRecorder)
}
```
any implementation of this interface will also be an implementation of the cluster interface. we'd need to rename the method in the interface to something more specific if we want that behaviour

Reply:
Yes you're right, complete brain failure there. Unless we had a different name for the method or something silly like that it won't help. `ReconcilerEvent` as an interface is fine then.